pkg/clients/cloudwatch/v1/input.go

```diff
@@ -59,18 +59,20 @@ func createGetMetricDataInput(getMetricData []*model.CloudwatchData, namespace *
 	}
 }
 
-func toCloudWatchDimensions(dimensions []*model.Dimension) []*cloudwatch.Dimension {
+func toCloudWatchDimensions(dimensions []model.Dimension) []*cloudwatch.Dimension {
 	cwDim := make([]*cloudwatch.Dimension, 0, len(dimensions))
 	for _, dim := range dimensions {
+		// Don't take pointers directly to loop variables
+		cDim := dim
```

Comment:
The code below doesn't take a pointer to a loop variable, it takes a pointer to a field in a struct, which behaves the same with & without this `cDim` copy: https://go.dev/play/p/WUzQblUn-2z. I removed a couple of similar copies in !1288 which turned out to have been unnecessary before & after the change.

It may be easier to prove that it's safe if YACE stops using explicit pointers and uses the AWS SDK's pointer conversion utilities ([v1](https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.50.20/aws#hdr-Value_and_Pointer_Conversion_Utilities), [v2](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2@v1.25.0/aws#hdr-Value_and_Pointer_Conversion_Utilities)), e.g. `aws.String(dim.Name)` and `aws.String(dim.Value)`. Those helpers are given a string as a value/copy, so they never need callers to make their own copies of the string variables.

Reply:
Apologies, this is my mistake. This behaviour changed in 1.22 (the default for go.dev/play), and the copy _is_ required in 1.21 and older for this pattern. Switching the dropdown to 1.21 on go.dev/play shows the wrong results.

Using `aws.String()` would avoid the copy in both versions, but isn't necessary if you use `cDim` here.

Reply:
Seems that the next golangci release should include some tooling to check all the unnecessary loop variables, I'll check on that once it's released.

Reply:
I'm sooooo happy they fixed this behavior in 1.22. If there wasn't a linter to catch it I would get burned by it constantly.